nixos/modules/services/networking/prosody.nix

```diff
@@ -792,6 +797,8 @@ in
 
       https_ports = ${toLua cfg.httpsPorts}
 
+      ${ cfg.extraConfig }
```

Comment:
Can you explain why this had to be moved before the muc definition?

Reply:
Not OP, but this makes impossible for the generic `extraConfig` to be interpreted as part of the MUC component config. It does make sense.

Reply:
From [the manual](https://prosody.im/doc/configure#overview):
> The configuration is divided into two parts. The first part is known as the "global" section. All settings here apply to the whole server, and are the default for all virtual hosts.
> The second half of the file is a series of VirtualHost and Component definitions. Settings under each VirtualHost or Component line apply only to that host.

When `cfg.extraConfig` is rendered after the muc definition then it only applies to that muc and not globally, which is not how it behaved before and probably not what you want.

Reply:
Thanks for the explanation, and adding it to the commit message as well. This is much better :+1: